scripts/iso-manager.sh

```diff
@@ -0,0 +1,154 @@
+echo "Options:"
+echo "-p/--product:     Path to product folder or ISO"
+echo "-d/--dry-run:     Run actions in dry run mode"
+echo "-l/--log-file:    path to log file"
+}
+
+# Exit with error code and message
+die() {
+    echo "$*" 1>&2 ;
+    exit 1;
+}
+
+# Run commands and log their output
+_log() {
+    if [ -n "$LOGFILE" ]; then
+        "$@" >> "$LOGFILE" 2>&1
+    else
+        "$@" >&2
+    fi
+}
+
+# Run commands and display/write the output to stdout/logfile
+_info() {
+    if [ "$DEBUG" == "info" ]; then
+        if [ -n "$LOGFILE" ]; then
+            "$@" >> "$LOGFILE" 2>&1
+        else
+            "$@" >&2
+        fi
+    else
+        "$@" >&/dev/null
+    fi
+}
+
+# helper function to set the current saltenv
+_set_env() {
+    if [ -z "$SALTENV" ]; then
+        SALTENV="metalk8s-$(salt-call --out txt slsutil.renderer \
+            string="{{ pillar.metalk8s.nodes[grains.id].version }}" \
+            | cut -c 8-)"
+    fi
+}
+
+# helper function to check for element in array
+containsElement () {
+  local e match="$1"
+  shift
+  for e in "$@"; do
+      [[ "$e" == "$match" ]] && return 0;
+  done
+  return 1
+}
+
+_add_products() {
+    # Use salt file.serialize merge require having full list
+    products=($(salt-call pillar.get metalk8s:products \
+        --out txt | cut -d' ' -f2- | tr -d '[],'))
+    for p in "$@"; do
+        if ! containsElement "$p" "$@"; then
+            products+=("$p")
+        fi
+    done
+    _log echo "Collecting products..."
+    _info echo "${products[@]}"
+    # build product list
+    product_list=${products[0]}
+    for i in "${products[@]:1}"; do
+        product_list+=,$i
+    done
+    _log echo "Updating bootstrap.yaml"
+    _info salt-call state.single file.serialize "$BOOTSTRAP_CONFIG" \
+        dataset="{'products': {'metalk8s': [$product_list]}}" \
+        merge_if_exists=True \
+        formatter=yaml \
+        show_changes=True \
+        test=$DRY_RUN
+    return $?
+}
+
+_configure_products() {
+    # Mount products
+    _log echo "Mount products..."
+    _info salt-call state.sls metalk8s.products.mounted \
+        saltenv="$SALTENV" test=$DRY_RUN
+    # Configure repos
+    salt_envs=$(salt-call --out txt slsutil.renderer \
+        string="{{ salt.metalk8s.get_products().keys() | join(' ') }}" \
+        | cut -d' ' -f2-)
+    [[ -z $salt_envs ]] && die "Cannot detect products envs"
+    for salt_env in $salt_envs; do
+        _log echo "Configuring product $salt_env..."
+        _info salt-call --local state.sls metalk8s.products.configured \
+            saltenv="$salt_env" \
+            pillar="{'metalk8s': {'endpoints': \
+            $(salt-call --out txt pillar.get metalk8s:endpoints | cut -c 8-)}}" \
+            test=$DRY_RUN
+    done
+    # Make the new version available
+    _log echo "Making new versions available"
+    _info salt-call state.sls metalk8s.products saltenv="$SALTENV" test="$DRY_RUN"
+}
+
+# Main
+[ $# -lt 1 ] && _usage
```

Comment:
Like described in the ticket, this script should support no argument and will basically just skip `add_products`